returnn/tf/util/data.py

```diff
@@ -234,7 +234,7 @@ def get_for_batch_ctx(self, batch, ctx, allow_none=False):
       if same_base.dyn_size_ext:
         assert batch == batch_base
         same_base.batch = batch
-        assert not same_base.dyn_size_ext.batch
+        assert not same_base.dyn_size_ext.batch or same_base.dyn_size_ext.batch == batch
```

Comment:
Is this okay?
Otherwise, `rem_dim.declare_same_as(resolved_dims[rem_dim_idx])` failed for me when `rem_dim` was just copied from the input (in the `[-1,1]` split).

Reply:
Yea I guess ok.
The whole code is a bit messy...